code/renderers/vue3/src/decorateStory.ts

```diff
@@ -46,25 +47,31 @@ export function decorateStory(
 ): LegacyStoryFn<VueRenderer> {
   return decorators.reduce(
     (decorated: LegacyStoryFn<VueRenderer>, decorator) => (context: StoryContext<VueRenderer>) => {
-      let story: VueRenderer['storyResult'] | undefined;
-
+      let story: VueRenderer['storyResult'] = { isNull: true };
       const decoratedStory: VueRenderer['storyResult'] = decorator((update) => {
         story = decorated({
           ...context,
           ...sanitizeStoryContextUpdate(update),
         });
+        const argsChanged =
+          update && update.args && Object.keys(update).length === 1 && !isVNode(story);
+        // TODO: this is a hack to avoid re-rendering the story when the args are not changed
+        // we should find a better way to do this
+        // i should get only update = { args: { ...context.args, text:... } } from the decorator and not the whole context
+        if (argsChanged) {
+          story = h(story, update.args);
+        }
         return story;
       }, context);
 
-      if (!story) {
-        story = decorated(context);
-      }
+      if (story.isNull) story = decorated(context);
 
       if (decoratedStory === story) {
         return story;
       }
-
-      return prepare(decoratedStory, h(story, context.args)) as VueRenderer['storyResult'];
+      const props = story.props ?? context.args;
```

Comment:
Why do we prefer story.props above context.args? This only works when `story` is a VNode right? And not component options.

Reply:
here we could use just context.args, but we need to check first if the story args has been changed by some decorator, in these case inner story props and context.args are not same.

in your story in you added a decorator like this 
export const InheritanceStory = {
  decorators: [
    (storyFn: PartialStoryFn, context: StoryContext) =>
      storyFn({ args: { ...context.args, text: `story ${context.args['text']}` } }),
  ],
  args: {
    text: 'starting',
  },
}
```

This only works when story is VNode exactly  cause you had to pick the  args changes from the { update } decorator function, and pass them to the normal story ( not yet VNode )